src/components/FileSelector/FileSelector.tsx

```diff
@@ -72,68 +73,127 @@ const DropTextWrapper = styled(Padbox)`
+      isDragDisabled = false,
+      isDisabled = false,
+      hasError = false,
+      size = FileSelectorSizes.fill,
+      multiple,
+      accept,
+      minFileSize,
+      maxFileSize,
+      maxFiles,
+      onFilesDrop,
+      onFilesAccepted,
+      onFilesRejected,
+      onFileDialogCancel,
+      onFileDialogOpen,
+      onDragEnter,
+      onDragOver,
+      onDragLeave,
+      validator,
+      className,
+      ...props
+    }: FileSelectorProps,
+    ref,
+  ) => {
+    const { error } = useLogger('FileSelector');
+    const { getRootProps, getInputProps, isDragActive, isFocused } =
+      useDropzone({
+        disabled: isDisabled,
+        noClick: isClickDisabled,
+        noDrag: isDragDisabled,
+        multiple,
+        accept,
+        minSize: minFileSize,
+        maxSize: maxFileSize,
+        maxFiles,
+        onDrop: onFilesDrop,
+        onDropAccepted: onFilesAccepted,
+        onDropRejected: onFilesRejected,
+        onFileDialogCancel,
+        onFileDialogOpen,
+        onDragEnter,
+        onDragOver,
+        onDragLeave,
+        validator,
+      });
+    const sizes =
+      size === FileSelectorSizes.area && 'width' in props && 'height' in props
+        ? {
+            width: props.width,
+            height: props.height,
+          }
+        : {};
+    const passedProps = omit(['width', 'height'], props);
 
-  if (isClickDisabled && isDragDisabled) {
-    error(
-      'Either one of or both "isClickDisabled" and "isDragDisabled" properties must be set to "false".',
-    );
-    return null;
-  }
+    if (isClickDisabled && isDragDisabled) {
+      error(
+        'Either one of or both "isClickDisabled" and "isDragDisabled" properties must be set to "false".',
+      );
+      return null;
+    }
+
+    if (isClickDisabled) {
+      return (
+        <FileSelectorWrapper
+          {...getRootProps({
+            $isDragActive: isDragActive,
+            $isFocused: isFocused,
+            $hasError: hasError,
+            $isDisabled: isDisabled,
+            $size: size,
+            $width: sizes?.width,
+            $height: sizes?.height,
+            paddingSize: SpaceSizes.sm,
+            ...passedProps,
+          })}
+          className={cls(CLX_COMPONENT, className)}
+        >
+          <input ref={ref} {...getInputProps()} />
+          <DropTextWrapper
+            $isCentered={size === FileSelectorSizes.area}
+            paddingSize={SpaceSizes.sm}
+            paddingType={PaddingTypes.squish}
+          >
+            <Text
+              size={TextSizes.md}
+              variant={
+                isDisabled ? TextVariants.secondary : TextVariants.primary
+              }
+            >
+              {dropLabel}
+            </Text>
+          </DropTextWrapper>
+        </FileSelectorWrapper>
+      );
+    }
+
+    if (isDragDisabled) {
+      return (
+        <div className={cls(CLX_COMPONENT, className)}>
+          <input {...getInputProps()} />
```

Comment:
There are several options that can get rendered but the `ref` got only the first